### PR TITLE
Adds a tutorial for automated messages

### PR DESCRIPTION
**Pull request by iAmWillShepherd**
Addresses the [task in Asana](https://app.asana.com/0/1200032447127702/1200784559147413) to draft a tutorial for event-triggered notifications by adding instructions for using automated messages.

### CODE REVIEW

**Comment by jfishman1** [comment]
IMHO, the main purpose of tutorials is to walk the person through start to finish and get to the "Aha" moment. They shouldn't need to contact Support or ask for help, the Tutorial should guide them through it leaving no room for interpretation or error. 

When following the ReadMe, I don't see the point of calling this a tutorial, might as well just point to the [Automated messages guide](https://documentation.onesignal.com/docs/automated-messages)

I'm not sure if this is going to be 1 of 3 tutorials for Event Triggered Notifications, but the asana ticket outlines 3 ways to do "ETNs".

**Comment by iAmWillShepherd** [comment]
@jfishman1 

> IMHO, the main purpose of tutorials is to walk the person through start to finish and get to the "Aha" moment. They shouldn't need to contact Support or ask for help, the Tutorial should guide them through it leaving no room for interpretation or error.

Can you be specific on what you think is missing or which steps leave room for interpretation?

> When following the ReadMe, I don't see the point of calling this a tutorial, might as well just point to the [Automated messages guide](https://documentation.onesignal.com/docs/automated-messages)

Maybe we have differing ideas of what a tutorial is—my definition: a transferral of knowledge through a set of instructions to accomplish a specific task; I think what I have here satisfies that definition. Perhaps some guidance on what you mean will be helpful?

> I'm not sure if this is going to be 1 of 3 tutorials for Event-Triggered Notifications, but the asana ticket outlines 3 ways to do "ETNs".

This is one of three. PRs should be tight in scope. If those other sections weren't independent, I would've added them in this PR. See PR #2 to track work for the notification API.

**Comment by jfishman1** [comment]
@iAmWillShepherd I agree with your definition of tutorial, but IMHO this is not a transfer of knowledge on how to setup a specific task of event triggered notifications, it is a less detailed guide on how to setup automated messages.

Approaching this as someone who doesn't understand OneSignal and wants to setup "event" triggered notifications, it's not clear to me how to achieve that.

What I think you are missing:

1. Define the "event"

The customer needs to understand what exactly the "event" is and how to get that data into OneSignal.

I think you should start with a hypothetical event (user did or didn't finish a flow in the app/site) and build the tutorial around that.

I would start with data tags to explain getting the "event" into OneSignal part. 
- How to capture events with data tags or what that would look like
- How do I use data tags to capture the event?
- What types of data tags should I use? 
- What options do I have? 
- How should I structure them?
- What are some best practices based on different events?

2. How to group users based on the "event"
Now that tags are added, how do I know which users have them?
What if the event can occur multiple times? How do I structure the segment so the user flows in and out each time they perform the event?

3. Then setup the automated message

Based on your current guide:
1 - Starting with Automated Message using "Subscribed Users". It doesn't explain why to use that segment. I don't think it should use that segment since it is not based on an "event".
How does the automated message work for events?
How often does it send to the user?
How quickly does it send to the user?
Why in step 7: "check if the user returns to the app delivery option"
- explanation of what the automated message delivery schedule options are and how they work


2 - Templates, says "dynamically generate the content of push notifications and emails" but how or why would I do that? What are best practices?
- explanation of how to use tags for personalized templates


3 - "Inspecting data tags" doesn't show how the data tags look in the dashboard, just how to get them. If a marketer needs to setup the segment, how do they know what they look like or what to do?

4 - Segment shows the "Create Segment" step but doesn't show the data tag filter being used with a data tags set in step 3
Customer may not know what the tags look like on that page or how to setup a segment with them.
If I am a marketer, how do I setup a segment with data tags?

It might be best to record yourself going through, end-to-end, how to setup event triggered notifications as reference.

**Comment by iAmWillShepherd** [comment]
@jfishman1, great feedback! I'll get to work on revisions.

**Comment by iAmWillShepherd** [comment]
@jfishman1 I've made some refinements to touch on some of the points you listed. I'm currently trying to organize the doc, but I'm unsure of the best order to present some of the concepts. Could you advise on this?

Fwiw, this draft is still incomplete; I'm mainly interested if I'm now on the right track.

File: README.md [hunk 1]
```diff
@@ -1 +1,196 @@
...
+
+There are three approaches for implementing these types of notifications:
+
+- Leverage **Segments** for sending messages.
+- Use the OneSignal Notification API.
+- Use a platform SDK.
+
+This tutorial series will cover the three methods outlined above, but first, I need to cover the building blocks that make this possible: events, segments, and automated messages.
+
+## Example Use-case
```
Comment (jmadler) [inline README.md:13]: I suggest swapping this and the next paragraph as it forms a more direct line of increasing depth of subject matter coverage throughout the document.  i.e. pyramid-shaped

File: README.md [hunk 2]
```diff
@@ -1 +1,196 @@
...
+```json
+{
+  "last_site_visit_date": ""  // the potential user has never landed on our site
+```
+
+**Step 2 -** We collect user details like name, email, etc from the event's context.
+
+```json
+{
+  "last_site_visit_dt": "",
```
Comment (jmadler) [inline README.md]: s/dt/date/

File: README.md [hunk 3]
```diff
@@ -1 +1,196 @@
...
+**Step 3 -** use data tag to indicate wherein the onboarding process the user is and when they got there. If our process has only two steps, we would choose either `onboarding_started` or `onboarding_complete` and mark the time for each step.
+
+```json
+{
+  "last_site_visit_date": "",
+  "first_name": "Rick",
+  "last_name": "Sanchez",
+  "email": "picklerick@gmail.com",
+  "onboarding_stage": "started",
+  "onboarding_start_dt: "1638209022859" // 'Mon Nov 29 2021 12:04:00 GMT-0600 (Central Standard Time)'
```
Comment (jmadler) [inline README.md]: missing a " at the end of the key

File: README.md [hunk 4]
```diff
@@ -1 +1,196 @@
...
+```
+
+**Step 2 -** We collect user details like name, email, etc from the event's context.
+
+```json
+{
+  "last_site_visit_dt": "",
+  "first_name": "Rick",
+  "last_name": "Sanchez",
+  "email": "picklerick@gmail.com"
```
Comment (jmadler) [inline README.md]: unless you own this email, can we put something@example.com?

File: README.md [hunk 5]
```diff
@@ -1 +1,196 @@
...
+4. Complete the _Message_ form and save
+   <img width="1438" alt="message template" src="https://user-images.githubusercontent.com/1715082/137006870-da9066f1-1140-4acb-9eee-f1b5903a23b5.png">
+
+## Automated Messages
+
+[Automated messages](https://documentation.onesignal.com/docs/automated-messages) enable you to push notifications or emails when a user enters a segment and are a key building block for implementing event-triggered notifications.
+
+### How to create an automated message
+
+We can create automated messages using The OneSignal Dashboard by following these steps.
```
Comment (jmadler) [inline README.md]: I think "the OneSignal dashboard" might be more appropriate so as to not imply some product subbranding.  See Capitalization Rules in the Content Style Guide

File: README.md [hunk 6]
```diff
@@ -1 +1,196 @@
...
+1. Navigate to _Messages_
+   ![automated-message-1](https://user-images.githubusercontent.com/1715082/136893452-2decb342-09e4-4e6a-aa15-724c3a92534e.png)
+
+2. Select _Automated_ from the submenu
+   ![automated-message-2](https://user-images.githubusercontent.com/1715082/136895432-3090eec7-ecfa-4dce-bded-1022faecaec2.png)
+
+3. Click the _New Automated Push_ button
+   <img width="1440" alt="Screen Shot 2021-10-11 at 11 28 10 PM" src="https://user-images.githubusercontent.com/1715082/136896871-6ebf5f79-dd5e-411c-9214-81d15bc61069.png">
+
+4. Select _Send to particular segment(S)_ option
```
Comment (jmadler) [inline README.md]: (S) should be (s)

File: README.md [hunk 7]
```diff
@@ -1 +1,196 @@
...
+
+## What Are Events
+
+An event is an action or occurrence recognized by the software that usually originates from user interaction.
+
+Common events you may want to respond to include
+
+- Cart abandonment
+- User achievement
+- New user signup
```
Comment (jmadler) [inline README.md:32]: We should either use punctuation at the end of list items on this doc or don't -- the style guide doesn't have an opinion.  However we should be consistent.

File: README.md [hunk 8]
```diff
@@ -1 +1,196 @@
...
+
+Since segments depend on data tags, we must first learn how to inspect their values.
+
+1. Navigate to _Audience_
+   <img width="1437" alt="audience-menu-selection" src="https://user-images.githubusercontent.com/1715082/137007996-c1c20dc2-e649-4f78-8b38-8c8388d12f34.png">
+2. Select _All Users_ from the submenu
+   <img width="1441" alt="users-submenu-selection" src="https://user-images.githubusercontent.com/1715082/137008242-8b0c87e8-1f14-4eae-84bb-7c01ffea3f03.png">
+3. Adjust the _Displayed columns_ to see your tags
+   <img width="1440" alt="displayed columns" src="https://user-images.githubusercontent.com/1715082/137007643-9c3993db-c8ee-4737-bed6-e03bf0d80e2a.png">
+
```
Comment (jmadler) [inline README.md:152]: We should note the benefit of doing so, since we're asking them to do something.

File: README.md [hunk 9]
```diff
@@ -1 +1,196 @@
...
+### How to creating a message template
+
+We can create a message template using the dashboard by following these steps.
+
+1. Navigate to _Messages_
+2. Select _Templates_ from the submenu
+3. Click the _New Push Template_ button
+4. Complete the _Message_ form and save
+   <img width="1438" alt="message template" src="https://user-images.githubusercontent.com/1715082/137006870-da9066f1-1140-4acb-9eee-f1b5903a23b5.png">
+
```
Comment (jmadler) [inline README.md:166]: Same here -- sum up the section with a bit on how the template will be used.

File: README.md [hunk 10]
```diff
@@ -1 +1,196 @@
...
+## Automated Messages
+
+[Automated messages](https://documentation.onesignal.com/docs/automated-messages) enable you to push notifications or emails when a user enters a segment and are a key building block for implementing event-triggered notifications.
+
+### How to create an automated message
```
Comment (jmadler) [inline README.md]: See Capitalization Rules in the Content Style Guide for all headings

File: README.md (hunk 7 above)
Comment (jmadler) [reply README.md:32]: I like colons at the end of lines with illustrations following, and periods at the end of others.

File: README.md [hunk 11]
```diff
@@ -1 +1,196 @@
 # Event-triggered Notifications Tutorial
+
+An event-triggered notification is an automated messages sent in response to an event.
+
+There are three approaches for implementing these types of notifications:
+
+- Leverage **Segments** for sending messages.
```
Comment (jmadler) [inline README.md]: Maybe I'm missing something, but this doesn't seem like three distinct techniques, this seems like: 
"Automating the Grouping of Users into Segments", "Tagging through the ... API", "Tagging in the ... SDK".  

In other words, segments seems like they stand out in this set, which suggests that this isn't a MECE list.  Is it a list of steps with options?  Give it some thought.

File: README.md (hunk 7 above)
Comment (iAmWillShepherd) [reply README.md:32]: This should be consistent now.

**Pull request closed without merging**